### PR TITLE
Assignments for lesson 03

### PR DESCRIPTION
**Pull request by Swt16**
Assignments for lesson 03.

### CODE REVIEW

File: students/ShinTran/lesson03/SlicingLab.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+# Returns a sequence with the first and last items exchanged
+def exchangeFirstLast(sequence):
+	length = len(sequence) - 1
+	first = sequence[0]
+	last = sequence[length]
+	mid = sequence[1:length]
+	if type(sequence) is tuple:
+		return (last,) + mid + (first,)
+	else:
+		return last + mid + first
```
Comment (natasha-aleksandrova) [inline students/ShinTran/lesson03/SlicingLab.py]: couple comments here:
* per pep8: function names and var names should be snake_case -> `exchange_first_last`
* your indentation needs to be exactly 4 spaces (no tabs)
* this slicing assignment should be purely slicing, not indexing. [start:stop:step] is slicing / [index] is indexing
* there should be no special logic for the type. same code should work for string, tuple or list (the case when using slicing)

File: students/ShinTran/lesson03/SlicingLab.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+# Returns a sequence with the middle third, then last third,
+# then the first third in the new order.
+def midLastFirst(sequence):
+	third = int(len(sequence)/3)
+	firstThird = sequence[:third]
+	midLast = sequence[third:]
+	if type(sequence) is tuple and (len(firstThird) == 1):
+		return midLast + firstThird
+	else:
+		return midLast + firstThird
```
Comment (natasha-aleksandrova) [inline students/ShinTran/lesson03/SlicingLab.py]: should be no special logic here 😉

File: students/ShinTran/lesson03/list_lab.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+print(series1)
+response = int(input("Type a number to return the index of the list > "))
+print(series1[response-1])
+series1 = ["Strawberries"] + series1
+print(series1)
+series1.insert(0,"Blueberries")
+print(series1)
+for i in range(0,len(series1)):
+	if (series1[i][0] == "P"):
+		print(series1[i])
```
Comment (natasha-aleksandrova) [inline students/ShinTran/lesson03/list_lab.py]: your loop should be:
```
for fruit in fruits:
    if fruit[0] == "P":
        print(fruit)
```

File: students/ShinTran/lesson03/mailroom.py [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+	d4 = ["Patricia Brown", 19184.81]
+	d5 = ["Robert Jones", 51227.53]
+	d6 = ["Jennifer Miller", 53514.94]
+	d7 = ["Michael David", 31051.46]
+	d8 = ["Mary Johnson", 71646.16]
+	d9 = ["William Rodriguez", 69244.21]
+	d10 = ["Mary Johnson", 12689.07]
+	d11 = ["Michael David", 57145.50]
+	d12 = ["Patricia Brown", 81679.15]
+	masterList = [d1,d2,d3,d4,d5,d6,d7,d8,d9,d10,d11,d12]
```
Comment (natasha-aleksandrova) [inline students/ShinTran/lesson03/mailroom.py]: why use anonymous vars?
```
donors = [
    ["James Smith", 342334],
    [Mary Johsnon", 3029843]
]
```
is more clear and pythonic

File: students/ShinTran/lesson03/mailroom.py [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+
+# Takes in a user input as a parameter, enters a donation, prints a report,
+# prints list, exit, prompts again if the input is bad
+# If the user types exit it'll go back to the main prompt
+def action(userInput):
+	tempList = []
+	createNameList()
+	if userInput == 'list':
+		for i in range(0,len(nameList)):
+			print(nameList[i])
```
Comment (natasha-aleksandrova) [inline students/ShinTran/lesson03/mailroom.py]: please review looping over lists; this pattern is in many places. looping over list:
```
for name in name_list:
    print(name)
```

**Comment by natasha-aleksandrova** [comment]
Per instructions on the website please post Pull Request link in the submission box. 
For this particular Pull Request the link is:

`https://github.com/UWPCE-PythonCert-ClassRepos/Self_Paced-Online/pull/813`

Simply copy the link from the browser bar after you create the Pull Request.

**Comment by Swt16** [comment]
I just reuploaded the mailroom assignment. I fixed the variables so it uses snake case format. I also updated the indentation from tabs to 4 spaces. For loops was also updated. I initially coded it in a certain way where it was failing for me, so I ended up using the for in range instead. I think I'm just too used to Java/C# style coding. Either way, Python is something that I wish to improve on!

**Comment by natasha-aleksandrova** [comment]
Understandable! If you use Pycharm (which is free and awesome) it comes with all the tools out of the box to encourage good python coding style 🙂

**Pull request merged**